src/lib/state/actions/query.js

```diff
@@ -52,10 +52,12 @@ export const onAppInitialized = (searchOnInit) => {
 };
 
 export const updateQueryState = (queryState) => {
-  return (dispatch) => {
+  return (dispatch, getState) => {
+    const {query: {size, page}} = getState()
+
     dispatch({
       type: SET_QUERY_STATE,
-      payload: queryState,
+      payload: {size, page, ...queryState},
```

Comment:
Question: would it be safer or more correct to this the following instead?
```suggestion
      const { query: currentQueryState } = getState()
      payload: {...currentQueryState, ...queryState},
```

Reply:
i think it in terms of safeness there is no big difference, because it both cases the new queryState object will override either the whole currentQueryState or just the page and size properties, if it has overlapping properties🤔